### DISCUSSION TITLE
scm_branch option

### DISCUSSION THREAD

**Discussion by ximwood** [ansibleguy76/ansibleforms#218] [Q&A] [answered]
Hello! Congratulations for the project! I have a question:
Is it possible to pass the scm_branch variable as an option in an AWX form? We need that variable not to be passed as an extra_var, but I haven't seen the option for the form in the documentation. We’ve noticed that variables like __template__ are passed as an option and can also be overridden as a field in the form. Could we pass a custom variable this way?

**Comment by ximwood** [comment, marked as answer]
Hello! I hope you are doing well.

First of all, I want to thank you again for the amazing work on this project! I’m following up on a question I asked about a week ago regarding the possibility of passing the scm_branch variable as an option in an AWX form, rather than as an extra_var. I understand that you’re likely very busy, but I wanted to kindly ask if there’s any guidance or suggestion you might be able to provide on this topic.

I would greatly appreciate any insight or help you could offer whenever you have the chance. Thank you so much in advance for your time and attention!

Best regards,

**Reply by ansibleguy76** [reply]
i must have missed this.

**Reply by ansibleguy76** [reply]
can you explain a little more?  is scm_branch a specific awx rest variable?

**Reply by ximwood** [reply]
In the configuration of an AWX project, we can set this option:
![image](https://github.com/user-attachments/assets/1037bdf4-f5cf-4de9-a3f5-25c8ae219015)
(allow branch override image)

Later, in the templates, we can specify that we want to use a particular branch:

![image](https://github.com/user-attachments/assets/767067a2-8084-4d2e-8282-083ee9bb910c)
(source control branch image)

Reviewing the API, this translates to the variable I mentioned:

![image](https://github.com/user-attachments/assets/c2956601-1cf7-4b7b-afd3-3192221df45c)
(scm_branch image api)


In the configuration of an AWX project, we can set this option:

Later, in the templates, we can specify that we want to use a particular branch:

Reviewing the API, this translates to the variable I mentioned:

The problem I'm facing is that if I add this variable as a field in the form, it gets added as an extra_var, but that's not how it should be. It should be outside of extra_vars, as an option at the same level as extra_vars, similar to the inventory or template.



I hope I've explained myself better now.

Just in case: https://docs.ansible.com/ansible-tower/latest/html/userguide/job_branching.html

**Reply by ansibleguy76** [reply]
Already added in the code.  will be in release 5.0.7

**Reply by ximwood** [reply]
Thank you so much!!!

**Reply by ansibleguy76** [reply]
transfered to issues
